platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileWriter.java

```diff
@@ -161,7 +162,7 @@ public static void writeSignedStateFilesToDirectory(
         writeEmergencyRecoveryFile(directory, signedState);
         if (!signedState.isGenesisState()) {
             // Genesis states do not have address books.
-            writeStateAddressBookFile(directory, signedState.getAddressBook());
+            writeStateAddressBookFile(directory, RosterUtils.buildAddressBook(signedState.getRoster()));
```

Comment:
Minor: we should probably file a ticket to revisit this part in the future because this method likely writes something in the form of a `config.txt` which is currently being deprecated. Ideally, we'd want to write a JSON of the Roster or something like that. But this may require evaluating current usages of this file and perhaps coordinating with DevOps/NodeOps/test frameworks that may use it.

Reply:
Created https://github.com/hashgraph/hedera-services/issues/17089